config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -155,39 +154,23 @@ public void deleteConfigs(
                 .asException());
         return;
       }
-      Map<ConfigResourceContext, Value> valuesByContext =
-          request.getConfigsList().stream()
-              .map(
-                  requestedDelete ->
-                      Map.entry(this.getConfigResourceContext(requestedDelete), emptyValue()))
-              .collect(ImmutableMap.toImmutableMap(Entry::getKey, Entry::getValue));
-
-      List<UpsertedConfig> deletedConfigs =
-          configStore.writeAllConfigs(valuesByContext, getUserId());
 
+      Set<ConfigResourceContext> configResourceContexts =
+          request.getConfigsList().stream()
+              .map(this::getConfigResourceContext)
+              .collect(Collectors.toUnmodifiableSet());
+      List<ContextSpecificConfig> configs = configStore.getConfigs(configResourceContexts);
```

Comment:
Is it possible to re-use `getAllConfigs#ResourceConfig`, instead of exposing the new function?

Reply:
I was actually confused between `getAllConfigs` and `getConfigs`. They looked the same based on the naming.

I understand the differentiation now